app/index.js

```diff
@@ -44,7 +44,10 @@ module.exports = generators.Base.extend({
         type: 'input',
         name: 'app_id',
         message: 'App ID',
-        default: answers => answers.name
+        validate: (input) => {
+          return (/\-|\s/.test(input)) ? 'Invalid App ID, use alphanumeric and periods only' : true;
```

Comment:
Yes, that makes sense. I agree that a list of Java keywords is out of scope. 

Looking at the regex you have there makes me feel that a unit test is now appropriate.
I intend to make several further project contributions, so I think now is the time to add testing.

If the roadmap includes and update to yeoman v2, then this will be especially useful.